### PR TITLE
feat: add txpool namespace RPC methods

### PR DESCRIPTION
**Pull request by crypto-facs**
Closes: #124 

## Description

Enabled txpool namespace with `txpool_content` method that complies with the go ethereum [documentation](https://geth.ethereum.org/docs/rpc/ns-txpool). The decision was to return an empty response (this is explained in further detail in the issue). Endpoint is needed to unblock block explorer integration

NOTE: Did not update documentation as the method is not fully functional yet. Happy to do it if it is required. 
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: ethereum/rpc/txpool_api.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package rpc
+
+import (
+	"github.com/cosmos/ethermint/ethereum/rpc/types"
+	log "github.com/xlab/suplog"
+)
+
+type PublicTxPoolAPI struct {
```
Comment (fedekunze) [inline ethereum/rpc/txpool_api.go:8]:
```suggestion
// PublicTxPoolAPI offers and API for the transaction pool. It only operates on data that is non-confidential.
type PublicTxPoolAPI struct {
```

File: ethereum/rpc/txpool_api.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	logger  log.Logger
+	backend Backend
+}
+
+func NewPublicTxPoolAPI(backend Backend) *PublicTxPoolAPI {
```
Comment (fedekunze) [inline ethereum/rpc/txpool_api.go:13]:
```suggestion
// NewPublicTxPoolAPI creates a new tx pool service that gives information about the transaction pool.
func NewPublicTxPoolAPI(backend Backend) *PublicTxPoolAPI {
```

**Review by fedekunze — COMMENTED** [review]
Looks good! Can you also add the `txpool_inspect` and `txpool_status` endpoints to return empty values? See https://geth.ethereum.org/docs/rpc/ns-txpool

**Comment by crypto-facs** [comment]
Hey @fedekunze thanks for the review! Just added the other two endpoints with a similar approach. Feel free to give it a final look!

File: ethereum/rpc/backend.go [hunk 3]
```diff
@@ -274,6 +279,33 @@ func (e *EVMBackend) PendingTransactions() ([]*types.RPCTransaction, error) {
...
+	return content, nil
+}
+
+// TxPoolStatus returns the number of pending and queued transaction in the pool.
+func (e *EVMBackend) TxPoolStatus() map[string]hexutil.Uint {
+	return map[string]hexutil.Uint{
+		"pending": hexutil.Uint(0),
+		"queued":  hexutil.Uint(0),
+	}
+}
```
Comment (fedekunze) [inline ethereum/rpc/backend.go:307]: nit, can you move these to `PublicTxPoolAPI` instead of `EVMBackend`?

**Review by fedekunze — APPROVED** [review]
ACK. Can you also add a description on de CHANGELOG.md with the issue number and description under the `Unreleased > Improvements` section?

**Comment by fedekunze** [comment]
follow up issue, add tests for RPC

**Pull request merged**